core/include/detray/utils/grid/detail/axis_bounds.hpp

```diff
@@ -87,6 +87,16 @@ struct open {
                             const std::size_t nbins) const noexcept {
         return map(range[0], range[1], nbins);
     }
+
+    /// Equality operator
+    ///
+    /// @param rhs the open axis to compare with
+    ///
+    /// @returns whether the two axes are equal
+    DETRAY_HOST_DEVICE
+    constexpr bool operator==(const open &rhs) const noexcept {
```

Comment:
I think, this operator is always true, because the type and the label are part of the class type (they are static constexpr variables).